hooks/open-telemetry/README.md

```diff
@@ -1,53 +1,97 @@
-run. It's recommended to register the `OpenTelemetryHook` globally in most situations, but it's possible to only enable
-the hook on specific clients. You should **never** register the `OpenTelemetryHook` both globally and on a client.
+OpenFeature provides various ways to register hooks. The location _where_ a hook is registered affects _when_ the hook is executed.
+It's recommended to register the hooks globally in most situations, but it's possible to only enable the hook on specific clients.
+
+> You should **never** register the OpenTelemetry hooks both globally and on a client.
+
+### TracesHook
+
+`TracesHook` provides OTel traces and export [span events](https://opentelemetry.io/docs/concepts/signals/traces/#span-events) for each feature flag evaluation.
+The hook implementation is attached to `after`and `error` [hook stages](https://github.com/open-feature/spec/blob/main/specification/sections/04-hooks.md#overview).
+
+Both successful and failed flag evaluations will add a span event named `feature_flag` with evaluation details such as flag key, provider name and variant.
+Failed evaluations can be allowed to set span status to `ERROR`. You can configure this behavior through`TracesHookOptions`.
+
+Consider the following code example for usage,
+
+```java
+final Tracer tracer=... // derive Tracer from OpenTelemetry
+
+// Set TracesHook globally
+OpenFeatureAPI.getInstance().addHooks(new TracesHook());
+
+// OpenFeature client
+final Client client=api.getClient();
+
+// Derive span from OTEL tracer
+Span span=tracer.spanBuilder("boolEvalLogic").startSpan();
+
+// Feature flag evaluation - Hook's span is derived from current context
+try(Scope ignored=span.makeCurrent()){
+    final Boolean boolEval=client.getBooleanValue(FLAG_KEY,false);
+}finally{
+    span.end();
+}
+```
+
+### MetricsHook
+
+`MetricsHook` performs metric collection by tapping into various hook stages.
+
+Below are the metrics extracted by this hook and dimensions they carry:
+
+| Metric key                             | Description                     | Unit         | Dimensions                                                    |
+|----------------------------------------|---------------------------------|--------------|---------------------------------------------------------------|
+| feature_flag.evaluation_requests_total | Number of evaluation requests   | {request}    | flag key & provider name                                      |
+| feature_flag.evaluation_success_total  | Flag evaluation successes       | {impression} | flag key, provider name, reason, variant & custom dimensions* |
+| feature_flag.evaluation_error_total    | Flag evaluation errors          | Counter      | flag key, provider name, exception                            |
+| feature_flag.evaluation_active_count   | Active flag evaluations counter | Counter      | flag key                                                      |
```

Comment:
Please update the dimensions once [this PR](https://github.com/open-feature/java-sdk-contrib/pull/359) is merged

Reply:
Updated